src/app/core/services/local-storage.service.ts

```diff
@@ -0,0 +1,23 @@
+import {Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LocalStorageService<T> {
+  saveData(key: string, value: T) {
+    localStorage.setItem(key, JSON.stringify(value));
+  }
+
+  getData(key: string): T | null {
+    const storedItem = localStorage.getItem(key);
+
+    return storedItem == null ? null : JSON.parse(storedItem);
+  }
+  removeData(key: string) {
+    localStorage.removeItem(key);
+  }
+
+  clearData() {
+    localStorage.clear();
+  }
+}
```

Comment:
Comment: the real threat when working with session/local storage are broken values, either from user intervention or version migrations, e.g. a string "idk" which is not part of the enum. No need to change anything now, but it is to be kept in mind. In practice the parsing operation needs to be more generic and fault tolerant, e.g. in our case a good parsing method would be "check if the value is in the enum values or default to the null or some value. No need to even risk a JSON parsing".